crates/oxc_linter/src/rules/unicorn/no_useless_undefined.rs

```diff
@@ -0,0 +1,687 @@
+use crate::{ast_util::is_method_call, context::LintContext, rule::Rule, AstNode};
+use oxc_ast::{
+    ast::{Argument, CallExpression, Expression, VariableDeclarationKind},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+fn warn() -> OxcDiagnostic {
+    OxcDiagnostic::warn(
+        "eslint-plugin-unicorn(no-useless-undefined): Do not use useless `undefined`.",
+    )
+    .with_help("Consider removing `undefined` or using `null` instead.")
+}
+fn no_useless_undefined_diagnostic(span0: Span) -> OxcDiagnostic {
+    warn().with_label(span0)
+}
+fn no_useless_undefined_diagnostic_spans(spans: Vec<Span>) -> OxcDiagnostic {
+    warn().with_labels(spans)
+}
+
+#[derive(Debug, Clone)]
+pub struct NoUselessUndefined {
+    check_arguments: bool,
+    check_arrow_function_body: bool,
+}
+
+impl Default for NoUselessUndefined {
+    fn default() -> Self {
+        Self { check_arguments: true, check_arrow_function_body: true }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Do not use useless `undefined`.
+    ///
+    /// ### Why is this bad?
+    /// `undefined` is the default value for new variables, parameters, return statements, etc… so specifying it doesn't make any difference.
+    ///
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // bad
+    /// let foo = undefined;
+    /// // good:
+    /// let foo;
+    /// ```
+    NoUselessUndefined,
+    correctness,
```

Comment:
Should be `pedantic`?

Reply:
It should be `pedantic`, because checking function parameters will cause false positives in some functions. like：compare function names 、array function